templates/install_puppet.bat.erb

```diff
@@ -6,7 +6,7 @@ FOR /F "tokens=*" %%A IN ('tasklist /FI "PID eq %AGENT_PID%" /NH') DO set _task=
 echo %_task% | findstr "No tasks are running" >nul
 IF NOT %errorlevel% == 0 ( GOTO wait_for_pid )
 
-msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*v "<%= @_logfile %>"
+start /wait msiexec.exe /qn /norestart /i "<%= @_msi_location %>" /l*v "<%= @_logfile %>"
```

Comment:
For reference, Beaker is always a good place to look to see how we install the Puppet MSI - https://github.com/puppetlabs/beaker/blob/f3d87f2b3463c2c49e1dd2b3e34b028336e7151e/lib/beaker/dsl/install_utils/windows_utils.rb#L18-L39

You've fixed the problem now with the `/wait` ... but I don't see the exit status being returned.